jib-core/src/main/java/com/google/cloud/tools/jib/api/Containerizer.java

```diff
@@ -94,8 +95,11 @@ public static Containerizer to(DockerDaemonImage dockerDaemonImage) {
         ImageConfiguration.builder(dockerDaemonImage.getImageReference()).build();
 
     DockerClient dockerClient =
-        new DockerClient(
-            dockerDaemonImage.getDockerExecutable(), dockerDaemonImage.getDockerEnvironment());
+        DockerClientResolver.resolve(dockerDaemonImage.getDockerEnvironment())
+            .orElse(
+                new CliDockerClient(
+                    dockerDaemonImage.getDockerExecutable(),
+                    dockerDaemonImage.getDockerEnvironment()));
```

Comment:
@eddumelendez could you please make it possible to pass an instance of `DockerClient` explicitly? (e.g. similar to `setExecutorService`, or maybe as an `to(DockerClient, DockerDaemonImage)` overload)